portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -16,28 +16,44 @@
 
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.RequestDispatcher;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.security.SecureRandom;
 
 @WebServlet(name = "HomeServlet",urlPatterns = {"/"})
 public class HomeServlet extends HttpServlet {
+DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
     response.setContentType("text/html");
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
+      // remove existing user with email 
       String userEmail = userService.getCurrentUser().getEmail();
+      int token = new SecureRandom().nextInt();
+      Entity userEntity = new Entity("User");
+      userEntity.setProperty("email", userEmail);
+      userEntity.setProperty("token", token);
+      userEntity.setProperty("timestamp",System.currentTimeMillis());
+      datastore.put(userEntity);
       String logoutUrl = userService.createLogoutURL("/");
-      response.sendRedirect("/frontPage.html?loggedIn=1"+"&url="+java.net.URLEncoder.encode(logoutUrl,"UTF-8")+"&email="+userEmail);
+      response.sendRedirect("/frontPage.html?email="+userEmail+"&url="+java.net.URLEncoder.encode(logoutUrl,"UTF-8")+"&token="+token);
```

Comment:
Now that I see, it we can prob do something like

```
string email = "";
string redirect_url = "";
int token = 0;

if (userService.isUserLoggedIn()) {
email = userEmail;
token = new SecureRandom().nextInt();
redirect_url = +java.net.URLEncoder.encode(login);
} else {
  email = "None";
  redirect_url = java.net.URL;
}

sendRedirect(String.format("/frontPage.html?email=%s&url=%s&token=%s", email, redirect_url, token));
```
This way we only call that method once